### PR TITLE
Updated Accumulo dependencies and classpath

### PR DESCRIPTION
**Pull request by mikewalch**
* Limited the number of dependencies from Hadoop
* Upgraded several dependencies including ZooKeeper,
  Jetty, Guava, Gson, Jersey, Bouncy Castle, ClassMate,
  FreeMarker, Caffeine, commons-cli, commons-codec,
  commons-configuration, commons-io, commons-lang,
  commons-logging, javax.ws-rs-api, HK2

### CODE REVIEW

**Comment by mikewalch** [comment]
Does anyone have objections if I merge this?

**Comment by ctubbsii** [comment]
I don't have any objections, but I did intend to review it, @mikewalch .

File: test/src/main/java/org/apache/accumulo/test/functional/ScannerIT.java [hunk 1]
```diff
@@ -98,7 +98,7 @@ public void testScannerReadaheadConfiguration() throws Exception {
       s.setBatchSize(1);
       s.setReadaheadThreshold(0L);
 
-      sw = new Stopwatch();
+      sw = Stopwatch.createUnstarted();
```
Comment (ctubbsii) [inline test/src/main/java/org/apache/accumulo/test/functional/ScannerIT.java]: Some of these could have made use of reset, but it's not a big deal.

File: assemble/conf/accumulo-env.sh [hunk 2]
```diff
@@ -51,14 +51,7 @@ if [ ! -d "$HADOOP_PREFIX" ]; then
...
 ## Add external Hadoop & Zookeeper dependencies to CLASSPATH.
-CLASSPATH="$(find "$ZOOKEEPER_HOME"/ "$HADOOP_PREFIX"/share/hadoop/{common,common/lib,hdfs,mapreduce,yarn} -maxdepth 1 -name '*.jar' \
-  -and -not -name '*slf4j*' \
-  -and -not -name '*fatjar*' \
-  -and -not -name '*-javadoc*' \
-  -and -not -name '*-sources*.jar' \
-  -and -not -name '*-test*.jar' \
-  -print0 | tr '\0' ':')$CLASSPATH"
-CLASSPATH="${conf}:${lib}/*:${HADOOP_CONF_DIR}:${CLASSPATH}"
+CLASSPATH="${conf}:${lib}/*:${HADOOP_CONF_DIR}:${ZOOKEEPER_HOME}/*:${HADOOP_PREFIX}/share/hadoop/client/*"
```
Comment (ctubbsii) [inline assemble/conf/accumulo-env.sh]: I think Hadoop actually went back to "HADOOP_HOME" as the preferred name in Hadoop 3. Not sure.

**Pull request merged**